primer-io/primer-sdk-ios
# feat: ApplePay Express Checkout

Logic to support Apple Pay Express Checkout.

Adds controls to `PrimerApplePaySettings` to mandate:
- Shipping Contact is present and retrieved
- Shipping Methods are required.

When `SHIPPING_MODULES` are present in the configuration, these will be surfaced to the user in the Apple Pay UI.
When a user changes their shipping contact, the `SHIPPING_MODULES` will be reevaluated to match the shipping address.

When `ShippingOptions` are not set in the ApplePaySettings, the existing Apple Pay flow should not be changed.

# Manual Testing

Test scenario [here](https://primer-io.gitlab.io/connections/qa-tools/test-scenario-docs/scenarios/apple-pay/express-checkout/)
Test plan [here](https://www.notion.so/primerio/Express-Checkout-testing-97e1a629b231420f9888281f72fa722b?pvs=4)
- Shipping addresses not in `GB` or `IE` should show as `unserviceable`
- Shipping addresses should show relevant shipping methods for the country

# Screenshots

_If applicable, otherwise remove this section_

# Contributor Checklist


# Reviewer Checklist


# Before Merging


# Other Stuff

- You can find out more about our automation checks [here](https://primerio.notion.site/iOS-Automation-Checks-198a1eb0e8994d999fb696d5902d97bb)
- Find out more about conventional commits [here](https://primerio.notion.site/Conventional-Commits-6ecfd6a0269a4db2af76d9f0537936b3)
